packages/backend/src/telemetry/service.ts

```diff
@@ -32,25 +32,29 @@ export interface TelemetryService {
   ): void
 }
 
-interface TelemetryServiceDependencies extends BaseService {
+export interface TelemetryServiceDependencies extends BaseService {
   instanceName: string
   collectorUrls: string[]
   exportIntervalMillis?: number
   aseRatesService: RatesService
   internalRatesService: RatesService
   baseAssetCode: string
   baseScale: number
+  enableTelemetry: boolean
 }
 
 const METER_NAME = 'Rafiki'
 
 export function createTelemetryService(
   deps: TelemetryServiceDependencies
 ): TelemetryService {
+  if (!deps.enableTelemetry) {
+    return new NoopTelemetryServiceImpl(deps)
+  }
```

Comment:
I _think_ I prefer having this `if (!deps.enableTelemetry)` check in `src/index.ts` instead of here, since it would seem clearer to me to understand how each class/dependency gets registered (kind of like tigerbeetle). Also wouldn't need to pass in `enableTelemetry: boolean`. NABD, just an option.

Reply:
that makes sense - good to have the switch as high level as possible I think.

Changed to return the noop or real one based on `config. enableTelemetry` in `src/index.ts`. In doing so I realized we don't need to pass in all the deps to the noop one so I removed those as well.